### PR TITLE
[clang][Interp] Implement IntegralAP subtraction

### PR DESCRIPTION
**Pull request by tbaederr**
The tests currently fail because they need one of the other open `IntegralAP` PRs.

Will update this once they are pushed.

### CODE REVIEW

File: clang/lib/AST/Interp/IntegralAP.h [hunk 1]
```diff
@@ -256,28 +255,23 @@ template <bool Signed> class IntegralAP final {
...
-  static bool CheckAddUB(const IntegralAP &A, const IntegralAP &B,
-                         unsigned BitWidth, IntegralAP *R) {
-    if (!A.isSigned()) {
-      R->V = A.V + B.V;
+  template <template <typename T> class Op>
+  static bool CheckAddSubUB(const IntegralAP &A, const IntegralAP &B,
+                            unsigned BitWidth, IntegralAP *R) {
+    if constexpr (!Signed) {
+      auto UOp = Op<APInt>();
+      R->V = UOp(A.V, B.V);
```
Comment (AaronBallman) [inline clang/lib/AST/Interp/IntegralAP.h:263]:
```suggestion
      R->V = Op<APInt>{}(A.V, B.V);
```

File: clang/lib/AST/Interp/IntegralAP.h [hunk 2]
```diff
@@ -256,28 +255,23 @@ template <bool Signed> class IntegralAP final {
...
     }
 
-    const APSInt &LHS = APSInt(A.V, A.isSigned());
-    const APSInt &RHS = APSInt(B.V, B.isSigned());
-
-    APSInt Value(LHS.extend(BitWidth) + RHS.extend(BitWidth), false);
+    auto SOp = Op<APSInt>();
+    const APSInt &LHS = A.toAPSInt();
+    const APSInt &RHS = B.toAPSInt();
+    APSInt Value(SOp(LHS.extend(BitWidth), RHS.extend(BitWidth)), false);
```
Comment (AaronBallman) [inline clang/lib/AST/Interp/IntegralAP.h:270]:
```suggestion
    const APSInt &LHS = A.toAPSInt();
    const APSInt &RHS = B.toAPSInt();
    APSInt Value(Op<APSInt>{}(LHS.extend(BitWidth), RHS.extend(BitWidth)), false);
```

**Review by AaronBallman — COMMENTED** [review]
Feel free to ignore the suggestions if you think they're not useful, but I tend to prefer not using a local variable for something used as a functor unless it's going to be reused.

Generally LG but I'd prefer to hold off on approval until CI is green with the changes.

**Comment by tbaederr** [comment]
Tests should work now

**Comment by tbaederr** [comment]
Ping

**Review by AaronBallman — APPROVED** [review]
LGTM!

**Pull request merged**